api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go

```diff
@@ -259,6 +261,11 @@ func (s *K8sStateProvider) Upsert(ctx context.Context, entry states.UpsertReques
 		if v, ok := dict["spec"]; ok && !entry.Options.UpdateStatusOnly {
 			item.Object["spec"] = v
 
+			resourceVersion := item.GetResourceVersion()
+			if entry.Value.ETag != "" {
+				resourceVersion = entry.Value.ETag
+			}
+			item.SetResourceVersion(resourceVersion)
```

Comment:
Discussed with Jiawei offline, we think we need to use entry.Value.Etag in resourceVersion even when it's empty